### ISSUE TITLE
how to capture a list of url with viewport and selector

### ISSUE THREAD

**Issue by brucejcw** [casperjs/casperjs#1467]
here is my code  

```
casper.start().each(views, function(self, view) {
    if (!view.viewport) {
      view.viewport = defaultViewport;
    }
    var image = 'screenshots/' + view.name + '-' + view.viewport.width + 'x' + view.viewport.height + '.png';
    self.then(function() {
      self.viewport(view.viewport.width, view.viewport.height);
    });
    self.thenOpen(view.url, function() {
      self.echo('Screenshot for ' + viewport.name + ' (' + view.viewport.width + 'x' + view.viewport.height + ')', 'info');
      self.captureSelector(image, view.selector) 
    });
  });
casper.run();
```

but no png created and no any log print

**Comment by kensoh** [comment]
Hi @brucejcw do you still have this issue? Do you see any possible cause if run with debug option? I haven't encountered issues with navigating to new URLs and then taking screenshots. Maybe it has been fixed.

**Comment by kensoh** [comment]
Hi @brucejcw not sure if this issue is resolved for you with latest CasperJS v1.1.4 and PhantomJS v2.1.1? Closing this for now but please reopen if it is still happening. Thanks for raising this!
